Readme.md

```diff
@@ -1,5 +1,137 @@
 # Flux
 
-lexer generator for any context free syntax in 100% rust
+A lexer generator for any context-free syntax in 100% rust with no dependencies.
 
-created for [Fender-lang](https://github.com/FenderLang/)
+Created for [Fender-lang](https://github.com/FenderLang/).
+
+# Lexer
+
+## What is a Lexer
+- A Lexer is part of an interpreter that takes sequence of characters and turns them into tokens.
+
+## How to use the Lexer
+
+First define your BNF input
+```rust
+let bnf_input = include_str!("../src/tests/bnf/fender.bnf");
+```
+
+Then you need to setup a matcher to be able to parse the bnf
+```rust
+    let mut lexer = match bnf::parse(bnf_input){
+        Ok(v) => v,
+        Err(e) => { 
+            //Error Messages
+            return;
+        },
+    };
+```
+
+## How to set Lexer Rules
+Now that you have `lexer` we can use that to now set some rules
+```rust
+ lexer.add_rule_for_names(vec!["pow", "add", "mul", "range", "cmp", "and", "or"], CullStrategy::LiftNChildren(1));
```

Comment:
This doesn't explain what the rules are at all. Without context this doesn't make any sense. It might help to include a snippet of example BNF, some examples of a rule or two you might want to set on the lexer, and an explanation of what they're doing and why. Removing whitespace tokens is a common use case to cover that's easy to understand.